asb-ballerina/src/asb/sender_connection.bal

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/java;
+
+public class SenderConnection {
+
+    handle asbSenderConnection;
+
+    private string connectionString;
+    private string entityPath;
+
+    # Initiates an Asb Sender Connection using the given connection configuration.
+    # 
+    # + connectionConfiguration - Configurations used to create a `asb:Connection`
+    public isolated function init(ConnectionConfiguration connectionConfiguration) {
+        self.connectionString = connectionConfiguration.connectionString;
+        self.entityPath = connectionConfiguration.entityPath;
+        self.asbSenderConnection = <handle> createSenderConnection(java:fromString(self.connectionString),
+            java:fromString(self.entityPath));
+    }
+
+    # Creates a Asb Sender Connection using the given connection parameters.
+    # 
+    # + connectionConfiguration - Configurations used to create a `asb:Connection`
+    # + return - An `asb:Error` if failed to create connection or else `()`
+    public isolated function createSenderConnection(ConnectionConfiguration connectionConfiguration) 
+        returns handle|Error? {
+        self.connectionString = connectionConfiguration.connectionString;
+        self.entityPath = connectionConfiguration.entityPath;
+        self.asbSenderConnection = <handle> createSenderConnection(java:fromString(self.connectionString),
+            java:fromString(self.entityPath));
+    }
+
+    # Closes the Asb Sender Connection using the given connection parameters.
+    #
+    # + return - An `asb:Error` if failed to close connection or else `()`
+    public isolated function closeSenderConnection() returns Error? {
+        return closeSenderConnection(self.asbSenderConnection);
+    }
+
+    # Send message to queue with a content and optional parameters
+    #
+    # + content - MessageBody content
+    # + parameters - Optional Message parameters 
+    # + properties - Message properties
+    # + return - An `asb:Error` if failed to send message or else `()`
+    public isolated function sendMessageWithConfigurableParameters(byte[] content, map<string> parameters,
+        map<string> properties) returns Error? {
```

Comment:
Can move some part of this line to the previous line.

Reply:
It exceeds